### DISCUSSION TITLE
Get a new follower / new sub event ?

### DISCUSSION THREAD

**Discussion by victordtrr** [TwitchIO/TwitchIO#396] [Questions]
I am already using the lib and now I want to be able to send a message in the chat when there is a new follower / new sub.

But I am not able so far to catch one of those two particular event. Can someone guide me a little bit please ? I tried several solutions that didn't work.

Regards.

**Comment by chillymosh** [comment]
Hi @victordtrr, in order to detect a new follower and sub, you will ideally need to use Eventsub.
Pubsub is easier and quicker to get going with but it does not do follows.

The downside (to the current implementation of Eventsub till websocket is officially released) is that it requires a publicly accessible domain, non self-signed SSL cert and a reverse proxy such as NGINX.

**Reply by victordtrr** [reply]
That's why I am not looking for the eventsub solution ... I think it's not possible in my case ?

**Reply by chillymosh** [reply]
Follows are only accessible via eventsub but subs are available from pubsub.
This will be easier once eventsub websockets is released, we do have a beta branch for this in 2.x for people to try but there are no models and everything is raw data currently.

The only other way is to poll the Helix API endpoints but then you'll need to keep a record of followers / subs which involves user oauth tokens, more hassle than eventsub if you aren't familiar with oauth.

**Reply by victordtrr** [reply]
I can parse rawdata. Can you provide me the beta branch ? I don't think it's currently listed.

If it's too complicated, I will parse the wizebot message in the event_message !

**Reply by chillymosh** [reply]
https://github.com/TwitchIO/TwitchIO/tree/feature/ws-eventsub
